### PR TITLE
feat: Add session type and providers

### PR DESCRIPTION
**Pull request by koji-1009**
## Description

* Add Session/Speaker/SessionData type
* Add sessionProvider/sessionsProvider/SessionDataProvider

Refs #17 #18 

## Type of change

Please delete options that are not relevant.

- [x] New feature

## How Has This Been Tested?

* Please describe the tests that you ran to verify your changes.
* Provide instructions so we can reproduce.

## Checklist

- [x] My code follows the style guidelines of this project.
- [x] My code has been formatted with `flutter format lib`.
- [x] My code has been fixed with `dart fix --apply lib`.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.

### CODE REVIEW

File: lib/model/sessions/session.dart [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+  factory Session.fromJson(Map<String, dynamic> json) =>
+      _$SessionFromJson(json);
+}
+
+enum SessionType {
+  talk,
+  sponsor,
+  event,
+}
```
Comment (Kurogoma4D) [inline lib/model/sessions/session.dart:46]: unionとenum両方で同じ分類があるのってどういう使い分けの想定でしたか？
個人的には片方で十分かなと思いました。

Comment (koji-1009) [reply lib/model/sessions/session.dart:46]: 確かに、sealed classのswitch分で十分かもですね。削除しておきます。

File: lib/model/sessions/session.dart [hunk 2]
```diff
@@ -7,7 +7,7 @@ part 'session.g.dart';
 @Freezed(unionKey: 'type')
 class Session with _$Session {
   const factory Session.talk({
-    required SessionType type,
+    required String type,
```
Comment (Kurogoma4D) [inline lib/model/sessions/session.dart:10]: freezedのunionKeyで指定されているキーはフィールドとして宣言しなくても自動でパースしてくれるので、typeごと不要かと思います。

Comment (koji-1009) [reply lib/model/sessions/session.dart]: [bb3fb8e](https://github.com/FlutterKaigi/conference-app-2023/pull/160/commits/bb3fb8ef2d6775e6dc88faeb67436fc8e4d21dc3) にて削除しました！

**Review by Kurogoma4D — APPROVED** [review]
LGTM👍

**Pull request merged**